### DISCUSSION TITLE
Web-based code editor in + for MicroPython

### DISCUSSION THREAD

**Discussion by vsolina** [micropython/micropython#9474] [Show and tell]
I made a code editor that is hosted on and runs on MicroPython+WiFi boards (tested it on ESP32) with a web interface where you can list directories, open text/py/html/js/css files, edit and store them. It also has an integrated WebREPL terminal to execute edited .py files.

Did not intend to publish this thing, but it is so useful for me that I wrote the majority of the editor code in the editor (it pulled itself by the bootstraps)

I hope someone else finds it as useful as I do: https://github.com/vsolina/micropython-web-editor

![interface](https://user-images.githubusercontent.com/7470243/193357232-b254f04d-db23-4cbe-9e26-6bdb64eb4ba2.png)

**Comment by solarkraft** [comment]
That's super cool!

**Reply by vsolina** [reply]
I'm glad you like it 🙌

**Comment by jczic** [comment]
You too :')
he he!!
